platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -0,0 +1,231 @@
+# Lifecycle management for the State
+
+## Summary
+
+This proposal describes a possible implementation of state lifecycle management that could work equally well for both the Consensus Node and the Block Node.
+
+|      Metadata      |                 Entities                 |
+|--------------------|------------------------------------------|
+| Designers          | [@imalygin](https://github.com/imalygin) |
+| Functional Impacts | Consensus Node, Block Node               |
+
+## Purpose and Context
+
+There is an initiative to create self-contained State modules (`swirlds-state-api` and `swirlds-state-impl`) that would have minimal dependencies, and specifically, they should not depend on `swirlds-platform-core`. Part of this initiative is the implementation of a state lifecycle management mechanism that could be used by both the Consensus Node and the Block Node.
+
+### Requirements
+
+- Add an interface representing state lifecycle management and its implementation.
+- Update the current State API to be coherent with this State Lifecycle API.
+- Implement this set of APIs and migrate the existing code to use it.
+
+## Design & Architecture
```

Comment:
I find myself wanting some summary of what the new functionality is at a level between the `Purpose and Context` and the `Design & Architecture` levels. Something that describes what this new feature will do before getting into the details like methods. What is meant by "lifecycle management" now that the client must maintain references?

Reply:
Somewhere, I'm not sure where but maybe here, it would be good to describe how the states flow through the CN along its lifecycle at a high level and how things will change with this new State API. For example, the fact that the CN will invoke the State Module to load a state from disk and to make fast copies when a round is done being handled, but still need to hold onto state objects with reservations to collect signatures. Then the equivalent for BN.

Reply:
I updated an existing diagram mapping the State method names to the part of the lifecycle. I also added a diagram that depicts how the final result is going to look like once we add `StateLifecycleManager`